File: 001/solution_001.R
#! /usr/bin/env Rscript

project.euler.001 <- function(n) {
  nums <- 1:(n-1)
  mults <- nums[nums %% 3 == 0 | nums %% 5 == 0]
  solution <- sum(mults)

  cat(solution, "\n")
}


project.euler.001(1000)

File: 001/solution_001.m
#! /usr/bin/env MathematicaScript -script

projectEuler001[upto_] := Module[
    { numbers
    , multiples
    , solution
    },
    numbers   = Range[0, upto - 1];
    multiples = Select[numbers, Mod[#, 3] == 0 || Mod[#, 5] == 0 &];
    solution  = Total[multiples];
    Print[solution];
];

projectEuler001[1000]
